### ISSUE TITLE
All layers disappear when moving window to another screen

### ISSUE THREAD

**Issue by ebuchlin** [Helioviewer-Project/JHelioviewer-SWHV#255]
When moving a JHelioViewer window from a screen to another screen, all layers (including grid, FOV, PFSS... and open observation layers) disappear. It is then possible to add again an observation layer, but grid, FOV, PFSS... are lost forever. Nothing should disappear in the first place.

See [attached screencast](https://github.com/Helioviewer-Project/JHelioviewer-SWHV/assets/3624676/552dfce5-0d59-454c-85cd-b5b260c25283)

System and JHV versions:

* Ubuntu Linux 23.10, xserver-xorg 1:7.7+23ubuntu2, plasma-framework_5.104.0a-0ubuntu1_amd64
* JHV/SWHV-4.5.2.10880 (amd64 Linux 6.2.0-35-generic) Eclipse Adoptium JRE 20.0.2 7.8GB 8CPU

**Comment by bogdanni** [comment]
I'm doing that for years in macOS, even between different pixel scaling displays (2x <-> 1x).
The fact the all layers are removed is very strange, could you please see if you can find something in Help->Show Log?

**Comment by ebuchlin** [comment]
I've had the issue for a long time but didn't have time or sufficient elements to report.

Here is the difference in logs between before and after:

```
2023-10-21 17:41:11.337 [AWT-EventQueue-0] SEVERE  org.helioviewer.jhv.opengl.GLInfo.checkGLErrors - GL Error (1281): invalid value - @GLListener.dispose()
2023-10-21 17:41:11.480 [AWT-EventQueue-0] INFO  org.helioviewer.jhv.opengl.GLInfo.get - OpenGL 4.6 (Core Profile) Mesa 23.0.4-0ubuntu1~23.04.1
```

**Comment by bogdanni** [comment]
Right, there's a catastrophic OpenGL error which kills all layers (1st line).
The canvas is reconfigured with a new OpenGL context (2nd line).

Even if it was possible to restore the layers, I have doubts that the image in the canvas can be reconstituted since the original context is lost.

Here's hoping that this bug is fixed in the underlying software (windowing system, Mesa, and/or JOGL).

**Comment by ebuchlin** [comment]
Temporary workaround: I can define the screen where I want JHV to appear as "primary screen". JHV will then open on that primary screen.

**Comment by bogdanni** [comment]
Can you please test the 5.0 build from https://swhv.oma.be/download_test/
I totally reworked the way GL rendering works. Do all the natural things you would do as a user (resize, move between screens, record movies, etc.)

If it works, can you please paste here the log line which starts with "ANGLE EGL config:" (log is via Help->Show Log).
